File: ruby/recursive_search.rb
def recursive_search(arr, target)
  if arr.length === 0
    return false
  end

  if arr[0] === target
    return true
  end

  arr.shift()
  recursive_search(arr, target)
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: true"
  puts "=>", recursive_search([1, 2, 3], 2)

  puts

  puts "Expecting: false"
  puts "=>", recursive_search([3, 2, 1], 4)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/recursive_search.js
function recursiveSearch(arr, target) {
  if (arr.length === 0) {
    return false;
  }

  if (arr[0] === target) {
      return true;
  }
  arr.shift();
  
  return recursiveSearch(arr, target);
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: true");
  console.log("=>", recursiveSearch([1, 2, 3], 2));

  console.log("");

  console.log("Expecting: false");
  console.log("=>", recursiveSearch([3, 2, 1], 4));
}

module.exports = recursiveSearch;

// Please add your pseudocode to this file

// 1. if array length is 0, return false. otherwise check the first index value of the array, if it is equal to target return true, otherwise use shift to remove first element and run the function again with the newly shortened array.

// And a written explanation of your solution
